### PR TITLE
fix /bin/wslpath missing

### LINKED ISSUE #104: wslpath symlink is missing
## Bug description
When I try to start vscode it complains that it cannot find `wslpath`.

## To Reproduce
```ShellSession
$ code .
/etc/profiles/per-user/nixos/bin/wslvar: line 177: wslpath: command not found
/mnt/c/Program Files/Microsoft VS Code/bin/code: line 41: wslpath: command not found
```

**Expected behavior**
There should be a binary in PATH that is named wslpath and is symlinked to /init

https://github.com/microsoft/WSL/issues/2715#issuecomment-386273515

### PR DESCRIPTION
**Pull request by nzbr**
Fixes #104

### CODE REVIEW

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
That only kinda works when doing ``PATH=$PATH:/bin code .``

NixOS does not include ``/bin`` into ``$PATH`` by default but adding it is also not necessary since wslpath only cares about the symlink name, not the path. So it should be enough to create a basic package which symlinks /init to ``$out/bin/wslpath``

**Comment by SuperSandro2000** [comment]
something like this

```nix
environment.systemPackages = [
(pkgs.writeShellScript "wslpath" ''
  ln -rs /init $out/bin/wslpath
'';)
];
```

File: modules/wsl-distro.nix [hunk 1]
```diff
@@ -129,6 +129,13 @@ with builtins; with lib;
           hosts.enable = false;
           "resolv.conf".enable = false;
         };
+
+        systemPackages = [
+          (pkgs.runCommand "wslpath" { } ''
+            mkdir -p $out/bin
+            ln -rs /init $out/bin/wslpath
```
Comment (SuperSandro2000) [inline modules/wsl-distro.nix:136]:
```suggestion
            ln -s /init $out/bin/wslpath
```

We don't want a relative symlink here.

File: modules/wsl-distro.nix [hunk 2]
```diff
@@ -152,15 +159,22 @@ with builtins; with lib;
...
+              mkdir -p /usr/share/$x
+              ${pkgs.rsync}/bin/rsync -ar --delete $systemConfig/sw/share/$x/. /usr/share/$x
+            done
+          ''
+        );
+        populateBin = stringAfter [ ] ''
+          ln -sf /init /bin/wslpath
+          ln -sf ${pkgs.bashInteractive}/bin/bash /bin/sh
+          ln -sf ${pkgs.util-linux}/bin/mount /bin/mount
+        '';
```
[the remark this answers is not in the document]
Comment (nzbr) [reply modules/wsl-distro.nix:176]: /bin/sh and /bin/mount get created when the rootfs tarball is created. They are needed for WSL to work correctly, so linking them to the version from the current nixpkgs is probably a good idea anyway. Having a /bin/wslpath probably won't hurt as well

[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply modules/wsl-distro.nix:176]: Yeah, true. Not that important tbh.

**Pull request merged**